packages/flutter/lib/src/widgets/platform_menu_bar.dart

```diff
@@ -0,0 +1,829 @@
+  /// platform.
+  static bool hasMenu(PlatformProvidedMenuItemType menu) {
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.android:
+      case TargetPlatform.iOS:
+      case TargetPlatform.fuchsia:
+        return false;
+      case TargetPlatform.linux:
+        return const <PlatformProvidedMenuItemType>{
+          PlatformProvidedMenuItemType.about,
+          PlatformProvidedMenuItemType.quit,
+        }.contains(menu);
+      case TargetPlatform.macOS:
+        return const <PlatformProvidedMenuItemType>{
+          PlatformProvidedMenuItemType.about,
+          PlatformProvidedMenuItemType.quit,
+          PlatformProvidedMenuItemType.servicesSubmenu,
+          PlatformProvidedMenuItemType.hide,
+          PlatformProvidedMenuItemType.hideOtherApplications,
+          PlatformProvidedMenuItemType.showAllApplications,
+          PlatformProvidedMenuItemType.startSpeaking,
+          PlatformProvidedMenuItemType.stopSpeaking,
+          PlatformProvidedMenuItemType.toggleFullScreen,
+          PlatformProvidedMenuItemType.minimizeWindow,
+          PlatformProvidedMenuItemType.zoomWindow,
+          PlatformProvidedMenuItemType.arrangeWindowInFront,
+        }.contains(menu);
+      case TargetPlatform.windows:
+        return const <PlatformProvidedMenuItemType>{
+          PlatformProvidedMenuItemType.about,
+          PlatformProvidedMenuItemType.quit,
+        }.contains(menu);
+    }
+  }
+
+  @override
+  Map<String, Object?> toChannelRepresentation(DefaultPlatformMenuDelegate delegate) {
+    if (!hasMenu(type)) {
+      throw ArgumentError(
+        'Platform ${defaultTargetPlatform.name} has no standard menu for '
+        '$type. Call StandardMenuItem.hasMenu to determine this before '
+        'instantiating one.',
+      );
+    }
+
+    return <String, Object?>{
+      _kIdKey: delegate.getId(this),
+      _kEnabledKey: enabled,
+      _kPlatformDefaultMenuKey: type.index,
+    };
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(FlagProperty('enabled', value: enabled, ifFalse: 'DISABLED'));
+  }
+}
+
+/// The list of possible standard, prebuilt menus for use in a [PlatformMenuBar].
+///
+/// These are menus that the platform typically provides that cannot be
+/// reproduced in Flutter without calling platform functions, but are standard
+/// on the platform.
+///
+/// Examples include things like the "Quit" or "Services" menu items on macOS.
+/// Not all platforms support all menu item types. Use
+/// [PlatformProvidedMenuItem.hasMenu] to know if a particular type is supported
+/// on a the current platform.
+///
+/// Add these to your [PlatformMenuBar] using the [PlatformProvidedMenuItem]
+/// class.
+///
+/// You can tell if the platform supports the given standard menu using the
+/// [PlatformProvidedMenuItem.hasMenu] method.
+// Must be kept in sync with the plugin code's enum of the same name.
```

Comment:
How was this list determined? How do we know it needs to be updated because Apple added another entry?

Reply:
It was determined by dumping the selectors for the existing default menu entries on a macOS app.  It's not really possible to detect it automatically, since we need to be able to make decisions about what should be platform provided, and what needs to be implemented in the app (in Dart). For instance, the default set includes Cut/Copy/Paste in the Edit menu, but those can't be implemented on the platform side, and must be implemented by the app (or the Framework).

I suppose I could write a test that checked the selectors of a default macOS app against our list, but it would probably mean automating XCode to generate the default app. It would probably make sense to have it be part of the process for updating our macOS shims.